shoop/admin/modules/products/views/forms.py

```diff
@@ -263,3 +284,23 @@ def save(self, commit=True):
 class ProductImageMediaFormSet(ProductMediaFormSet):
     allowed_media_kinds = [ProductMediaKind.IMAGE]
     form_class = ProductImageMediaForm
+
+    def save(self, commit=True):
+        """
+        Add first saved image as  primary image for this product by default if none is selected as such
+        """
+        super(ProductImageMediaFormSet, self).save(commit)
+
+        if not self.forms:
+            return
+
+        has_primary = False
+        for form in self.forms:
+            if form.cleaned_data.get("is_primary"):
+                has_primary = True
+                break
```

Comment:
Though then the test below would need to be amended to "if self.forms and not has_primary", so the 294-295 if reasonable.